fsio.go

```diff
@@ -114,7 +121,18 @@ func (fs *FileSystem) Open(p Path, offset, length int64, buffSize int) (io.ReadC
 	}
 
 	req, _ := http.NewRequest("GET", u.String(), nil)
-	rsp, err := fs.client.Do(req)
+
+	var rsp *http.Response
+	for retries := fs.Config.MaxRetries; retries > 0; retries-- {
+		// don't scope err in this loop
+		rsp, err = fs.client.Do(req)
+		if err == nil {
+			break
+		}
+		if t, ok := err.(isTemp); !ok || !t.Temporary() {
+			break
+		}
+	}
```

Comment:
@abraithwaite Maybe it would be good to put some sort of back-off logic to wait before next retry?